ibverbs/src/lib.rs

```diff
@@ -368,33 +372,53 @@ impl Context {
     /// A protection domain is a means of protection, and helps you create a group of object that
     /// can work together. If several objects were created using PD1, and others were created using
     /// PD2, working with objects from group1 together with objects from group2 will not work.
-    pub fn alloc_pd(&self) -> Result<ProtectionDomain, ()> {
+    pub fn alloc_pd(self: Arc<Self>) -> Result<ProtectionDomain, ()> {
         let pd = unsafe { ffi::ibv_alloc_pd(self.ctx) };
         if pd.is_null() {
             Err(())
         } else {
-            Ok(ProtectionDomain { ctx: self, pd })
+            Ok(ProtectionDomain {
+                ctx: self.clone(),
+                pd,
+            })
+        }
+    }
+
+    /// Query corresponding device
```

Comment:
This could definitely use some more docs :)

Reply:
Makes sense. Let me see how to clean up